docs/spec.md

```diff
@@ -140,11 +140,26 @@ which may allow us to remove tuple types from StableHLO
 
 ```ebnf
 ElementType ::= BooleanType | IntegerType | FloatType | ComplexType
+              | QuantizedType
 BooleanType ::= 'i1'
 IntegerType ::= 'si4' | 'si8' | 'si16' | 'si32' | 'si64'
               | 'ui4' | 'ui8' | 'ui16' | 'ui32' | 'ui64'
 FloatType   ::= 'f8E4M3FN' | 'f8E5M2' | 'bf16' | 'f16' | 'f32' | 'f64'
 ComplexType ::= 'complex' '<' ('f32' | 'f64') '>'
+QuantizedType ::= '!quant.uniform' '<' QuantizationStorageType
```

Comment:
If I read this correctly, we are using different syntax w.r.t. TFLite.  Is this intended?

For example:

```
TFL: !quant.uniform<u8<0,255>:f32, 1.0:0>
HLO: !quant.uniform<ui8:f32:0:255:[1.0]:[0]>

TFL: !quant.uniform<u8<0,255>:f32:3, {1.0:0, 2.0:2}>
HLO: !quant.uniform<ui8:f32:0:255:[1.0, 2.0]:[0, 2]:3>
```

Where `storage_min` and `storage_max` are not optional, scales are grouped together, and `zero_points` are grouped together.

BTW, the unification of signed flag and storage type is good IMO.

Reply:
>> If I read this correctly, we are using different syntax w.r.t. TFLite. Is this intended?


Yes, you are right in pointing the differences.
 
While we were speccing the StableHLO opset, we made a conscious decision about not to use MLIR or XLA as the notational vehicle for the spec. I think the reason is to keep the specification immune to any future formatting changes. As an example, in the specification, MLIR attributes are called constants.

I believe the same decision has been extended here to TFL specific notations.

Reply:
>> we made a conscious decision about not to use MLIR or XLA as the notational vehicle for the spec

Realized that the decision is more specific to using MLIR notations like Traits, Attributes etc in the specification.  A grammar that follows MLIR closely could benefit from familiarity.

Updated the ebnf grammar together with relevant changes in constraints.
PTAL.